lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
```

Comment:
@purbon could it use the subject directly instead of using the described class and the config?
I am just saying that because on the plugin I am updating it used `options` instead of config in the test.

maybe something like

```ruby
    context "when interrupting the plugin" do
      subject { MyPLugin.new(config) }

      it_behaves_like "an interruptible input plugin"
    end
```